store/store.go

```diff
@@ -18,13 +16,20 @@ import (
 // for keeping the application state in sync with the database.
 type Store struct {
 	*models.ORM
-	Config      Config
-	Clock       AfterNower
-	Exiter      func(int)
-	KeyStore    *KeyStore
-	TxManager   *TxManager
-	HeadTracker *HeadTracker
-	sigs        chan os.Signal
+	Config    Config
+	Clock     AfterNower
+	Exiter    func(int)
+	KeyStore  *KeyStore
+	TxManager *TxManager
+	sigs      chan os.Signal
+}
+
+type rpcSubscriptionWrapper struct {
+	*rpc.Client
+}
+
+func (wrapper rpcSubscriptionWrapper) EthSubscribe(ctx context.Context, channel interface{}, args ...interface{}) (models.EthSubscription, error) {
```

Comment:
👍

Reply:
On second thought, I've heard that best practice in Go is to "[accept interfaces and return structs]{http://idiomaticgo.com/post/best-practice/accept-interfaces-return-structs/}." It looks like we're already accepting EthSubscriptions in other places. Any reason we need to return an interface here?

Reply:
To comply with the `CallerSubscriber` interface, which returns an `EthSubscription` interface because it can either be a `MockSubscription` or `rpcSubscriptionWrapper `. We can't return a concrete type here because the implementation can vary in tests.

```golang
type CallerSubscriber interface {
	Call(result interface{}, method string, args ...interface{}) error
	EthSubscribe(context.Context, interface{}, ...interface{}) (models.EthSubscription, error)
}
```

https://github.com/smartcontractkit/chainlink/blob/features/periodically-reconnect-to-eth-node/store/eth_client.go#L23-L26

Reply:
I see. Thanks for the clarification.